gazebo_plugins/package.xml

```diff
@@ -52,4 +53,7 @@
 
   <test_depend>rostest</test_depend>
 
+  <export>
+    <gazebo_ros plugin_path="${prefix}/../../lib" gazebo_media_path="${prefix}" />
```

Comment:
This looks a bit fishy to me. What kind of workspace have you tried it on? The [usual pattern](http://gazebosim.org/tutorials/?tut=ros_plugins#Updatepackage.xml) is:

`plugin_path="${prefix}/lib"`

Reply:
I'm using [Nix](https://github.com/lopsided98/nix-ros-overlay), which tends to expose edge case bugs like this because it installs each package into its own directory.

AFAICT, this patch is correct and the documentation is wrong. I opened a PR to fix the documentation: https://bitbucket.org/osrf/gazebo_tutorials/pull-requests/539

`${prefix}` refers to the directory containing the `package.xml` file. This is implemented in rospkg (https://github.com/ros-infrastructure/rospkg/blob/master/src/rospkg/manifest.py#L359) and rospack (https://github.com/ros/rospack/blob/master/src/rospack.cpp#L2186, https://github.com/ros/rospack/blob/master/src/rospack.cpp#L1399).

Reply:
Thank you for digging in, I believe you're right. I checked that a few installation paths match your pattern:

On Ubuntu from debs:

```
/opt/ros/melodic/share/gazebo_plugins/package.xml
/opt/ros/melodic/lib/libgazebo_ros_imu.so
```

On a colcon workspace with `--merge-install`:

```
install/share/gazebo_plugins/package.xml
install/lib/libgazebo_ros_imu.so
```

On a colcon workspace without `--merge-install`:

```
install/gazebo_plugins/lib/libgazebo_ros_imu.so
install/gazebo_plugins/share/gazebo_plugins/package.xml
```

> I'm wondering how this hasn't come up before.

I believe the plugins are picked up because the `lib` path is included in `LD_LIBRARY_PATH` when sourcing `setup.sh`. But you're right that without this PR they're not picked up by `ros::package`.

---

(FYI, a couple of `master` on your links above should be changed to `melodic-devel`)